### ISSUE TITLE
[REQUEST] - Making Items that are capable of holding other items

### ISSUE THREAD

**Issue by p4535992** [fantasycalendar/FoundryVTT-ItemPiles#120]
**Is your feature request related to a problem? Please describe.**

No problem is related to this request.

**Describe the solution you'd like**

The concept is a relatively simple Items that are capable of holding other items.

The idea is the same as the form https://gitlab.com/tposney/itemcollection/, but if you could integrate the concept of item piles as a "bag of items" on the item document instead of on the document of the Actor, it would be a multi-system solution of that module.

The priority is very low and I don't know if it is of interest to others as Dnd5e and Pf2e already have mechanisms of this kind integrated in the base system or with other modules.

Very Optional: It would be interesting to have the possibility to share the content of the item with other players (as happens now for the tokens on the canvas with active item piles) to emulate absurd things like the "Many-Handed Pouch" artificier infusion.

**Describe alternatives you've considered**

Item Collection for the system Dnd5e

**Additional context**

**Comment by gsterling** [comment]
I am also very interessted in this system agnostic solution.

**Comment by Haxxer** [comment]
Thanks for the suggestion, a few questions;

- Considering every system has its own actor inventory, how would the module reliably allow people to open the item container?
- In addition, each system has its own item types, some of which may not have a "container" type, how would that be handled?

**Comment by p4535992** [comment]
Starting from the fact that you are much more experienced than me, this is what I thought:

_Considering every system has its own actor inventory, how would the module reliably allow people to open the item container?_

I believe that we can avoid thinking in terms of actors and inventory management, I would work only and exclusively on the item document. My idea was to save on the flags of the item document (if enabled by Item Piles) the list of ItemData of the items "contained", click the edit button and rendering a specific html template for displaying the list of objects, modifying the drag and drop management  (perhaps even the same template already used by Item Piles on actors), values such as "total weight",  and "sum all prices" of the item container are more "articulated" and must be studied system by system, i don't need these, but maybe someone else yes ? For me is not very important.

_In addition, each system has its own item types, some of which may not have a "container" type, how would that be handled?_

Here I had thought of two ways, but the important thing is not to create a new type of object.

The first option is to "choose" for each system a default type to be set (for example the type backack on dnd5e as item collection does now).
The second option ignores the type of the object and is enabled, any type of object by clicking on the "Item Piles" button on the item sheet header (similar to how it is done now for the actors) and everything is managed with flags for save the data and an html template with listeners to perform the various actions (edit, delete, insert, ecc.).

the second is the one I like best because it is much more generic.

In conclusion, as I see it, the data are saved on the flags of the item document of the container item, the rest is managed by an html template as generic as possible, with a simple list view of the objects present in the container.

**Comment by Orionox** [comment]
I'd also be very interested in this sort of improvement. Currently, I use Item Collections, but it has some oddities that are proving frustrating in v10. Also, it would be nice if item piles were like the one-stop-shop for all your item management needs, and adding this functionality would allow me to remove item collections.

**Comment by p4535992** [comment]
Another interesting approach is given from krbz999  and the module [backpack manager](https://github.com/krbz999/backpack-manager/), still limited to dnd5e, but the concept to use the actor document instead the item document maybe is more "in line" with the item piles logic.

**Comment by EternalRider** [comment]
Also want such a system independent function. And I also hope that this container can be thrown to the ground and presented as item-pile, my idea is to make this container can imply the actor data of an item-pile, and use the same mechanism as opening the item-pile when opening the container, And this item-pile actor can be created when the container is dropped on the map.

**Comment by p4535992** [comment]
The ticket is closed, because in these months i see things and i can tell now the solution is "not inherent to the module" very simply the container type objects must be managed by the system itself, Dnd5e (next version 2.1.X), Pf2e , Pf1e, ecc. already have their own "container" system and at this point I think it is the most viable solution.
